### ISSUE TITLE
I can't use Fanbox f3 (nor f1 and f2 anymore)

### ISSUE THREAD

**Issue by krax11** [Nandaka/PixivUtil2#1386]
### Prerequisites

* [ ] Did you read FAQ section in [readme.md](https://github.com/Nandaka/PixivUtil2/blob/master/readme.md)?
* [ ] Did you test with the latest [releases](https://github.com/Nandaka/PixivUtil2/releases "latest releases") or [commit](https://github.com/Nandaka/PixivUtil2/commits/master "latest commit") ?
* [ ] Did you search for existing issues in [Issues](https://github.com/Nandaka/PixivUtil2/issues)?

### Description

f1-f3 doesn't work for me. The last 2 months I was using f3 because f1 and f2 weren't working, but now it doesn't work anymore.

### Steps to Reproduce

1. I try to use f1 to f3 in my PixivUtil2 folder where I download Fanbox images, using the latest release (normal realease or Master).
2. Retrieving images, zips or any type of media fails.
3. I get an error related to decode.

**Expected behavior:** It downloads the individual posts or from supporting list or individual artists.

**Actual behavior:** Input: f1
End Page (default is 0) = 0
Trying to log in FANBOX with saved cookie
Traceback (most recent call last):
  File "PixivUtil2.py", line 1745, in main
  File "PixivUtil2.py", line 1467, in main_loop
  File "PixivUtil2.py", line 920, in menu_fanbox_download_from_list
  File "PixivBrowserFactory.py", line 940, in fanboxGetArtistList
    self.fanbox_is_logged_in()
  File "PixivBrowserFactory.py", line 447, in fanbox_is_logged_in
    if not self.fanboxLoginUsingCookie(self._config.cookieFanbox):
  File "PixivBrowserFactory.py", line 423, in fanboxLoginUsingCookie
    if '"user":{"isLoggedIn":true' in str(parsed.decode('utf-8')):
AttributeError: 'str' object has no attribute 'decode'
Unknown Error, please check the log file: (<class 'AttributeError'>, AttributeError("'str' object has no attribute 'decode'"), <traceback object at 0x000001D1E9350600>)

It always fails in the normal release and in the Master release becuase of the decode.

I changed the language to japanese on Pixiv and Fanbox, changed the specific lines where pixiv.net/en was referenced. I tried to do everything I read in the open issues related to this, but nothing works...

**Log file:** 
[pixivutil.log](https://github.com/user-attachments/files/17599625/pixivutil.log) Normal realease version
[changelog.txt](https://github.com/user-attachments/files/17599634/changelog.txt) Master version

### Versions

You can get this information from executing `PixivUtil2.py --help`.
[pixivutil.log](https://github.com/user-attachments/files/17599643/pixivutil.log)

**Comment by gnarf1975** [comment]
Same Problem here. 
**And I'm just a user, not a programmer, tinkerer or anything like that. So I would be very grateful for a proper update.**

Thank you very much 
(Donator)

**Comment by biggestsonicfan** [comment]
The update is the dependencies, not PixivUtil2 itself. There is no programming involved.

**Comment by BillyFang** [comment]
Hi, I did the update per suggestion as shown below
![2024-12-04 22_00_06-Local Disk (C_)](https://github.com/user-attachments/assets/f42374fc-3434-40c9-b20d-d425041c5887)
![2024-12-04 22_00_45-Administrator_ Command Prompt](https://github.com/user-attachments/assets/4ce1c02a-b9f1-41e0-bb2b-f034cc88ae97)
I then ran the tool, but it still gave me the same error. 
![2024-12-04 22_03_33-Select PixivDownloader 20240703 1 of 1 FanboxArtist, pa](https://github.com/user-attachments/assets/0cd24630-8ffc-447e-9ebc-cf44a8637d51)
Here's the log below.
[pixivutil.log](https://github.com/user-attachments/files/18007001/pixivutil.log)

**Comment by LordBarsdale** [comment]
I mean, I get his frustration. A "proper update" in this context would mean launching the program would automatically update your installed required packages for you, or tell you to do so and how in layman's terms.

**Comment by biggestsonicfan** [comment]
@LordBarsdale 
> I mean, I get his frustration. A "proper update" in this context would mean launching the program would automatically update your installed required packages for you, or tell you to do so and how in layman's terms.

I've always thought a proper update in their eyes meant an update to the exe, not just the python scripts.

In any case, I will be testing PixivUtil2 on Windows today to see if I can replicate these issues.

@BillyFang
> Hi, I did the update per suggestion as shown below

Your log shows `C:\Users\nchek\Dropbox\PixivUtil\dist\PixivUtil2\PixivUtil2.exe`, the exe uses it's own baked in dependencies, and therefore it will not solve the problem if you update the pip dependencies. You will need to use the python script in order to resolve the issue.

**Comment by LordBarsdale** [comment]
Need to VPN to a japanese IP

**Comment by BillyFang** [comment]
> Your log shows `C:\Users\nchek\Dropbox\PixivUtil\dist\PixivUtil2\PixivUtil2.exe`, the exe uses it's own baked in dependencies, and therefore it will not solve the problem if you update the pip dependencies. You will need to use the python script in order to resolve the issue.

@biggestsonicfan
Are you referring to the PixivUtil2.py script from the master file folder?
I tried running it, then it requested my logins.
![2024-12-08 13_30_37-Select PixivDownloader 20240703](https://github.com/user-attachments/assets/01648bcc-3fe7-4657-b2e9-b802d40bff75)

Then I got new errors where it's not reading the logins.
![2024-12-08 13_34_38-Select PixivDownloader 20240703](https://github.com/user-attachments/assets/ad78d878-e043-4433-a7fe-adb1b6076d4c)

Note, the PixivUtil2.py script is in a separate folder but the wiki doesn't seem to say you need it in a specific spot afaik.

**Comment by biggestsonicfan** [comment]
@BillyFang 
> Then I got new errors where it's not reading the logins.

Correct, this is a _different_ error. If you can get past it, fanbox will work.

**Comment by gnarf1975** [comment]
**This Sucks**

As a **Normal User with no experience in this Shit**, I will probably be able to delete this nice tool at some point because it has become unusable. 

There used to be a regular update at the beginning, today the smart users only discuss, and the rest is just out of luck. 

☕

**Comment by NHOrus** [comment]
Look, every time Nandaka fixed something I requested, I made a donation, until I was unable to.
I frankly suggest everyone does the same, because fixing this utility is thankless, horrible duty, especially with naysayers who want it all for free.

**Comment by SSoM** [comment]
> Try updating your installed required packages `pip install --upgrade --force-reinstall -r requirements.txt`

Where? CMD gives me this error even as Administrator.
ERROR: Could not open requirements file: [Errno 2] No such file or directory: 'requirements.txt'

**Comment by SSoM** [comment]
Opted for the commit/master version, but now I can't log in. Exact same issue as Billyfang. Latest Python 313. How do I get past this error? I'm certain I'm using the correct cookies, considering they work on the config of the exe version. Even logging out and back in with new cookies didn't work.

**Comment by LordBarsdale** [comment]
> **This Sucks**
> 
> As a **Normal User with no experience in this Shit**, I will probably be able to delete this nice tool at some point, because it has become unusable.
> 
> ☕

I am also a normal user, you need to do several things differently to get fanbox working properly. Find half an hour of time, sit down with some tea, and get ready to use a command window and type stuff. I AM NOT an expert or even familiar with this stuff. Please take this all with a grain of salt and correct me where I get things wrong. 

First, you need to update the requirements as listed above. This means open cmd or powershell and running the commands listed above here on this page. if you get an error, just google the error. In my case, I had to run a command to uninstall the old ones I think, manually paste PixivUtil2's requirement.txt list the windows folder where your computer looks for it, then run the install command again until it cleared. I was just googling stuff and trying things listed here until it worked. Take a second and try doing that until you get the requirements updated.

https://github.com/Nandaka/PixivUtil2/blob/master/requirements.txt
Download this file obviously.

You can't use the exe in the normal release, you have to use:
https://github.com/user-attachments/files/16531982/PixivUtil.zip


Here's how:
https://github.com/Nandaka/PixivUtil2/issues/1361#issuecomment-2275990761
Unzip the folder, open it. You should have a folder named PixivUtil2-20240703. This will be where you actual launch the program for now. Forget the old one until we get a proper update.

Make sure you have python installed. Google python and install it.

Replace the two files listed here in the folder named PixivUtil2-20240703
https://github.com/Nandaka/PixivUtil2/issues/1361#issuecomment-2273943969

AND make these changes listed here to the same folder
https://github.com/Nandaka/PixivUtil2/issues/1352#issuecomment-2227002669

Then go to the address window of that folder you just downloaded, unzipped, and have been editing this whole time named PixivUtil2-20240703. You see the address window where normally it's just the drive location (E:/User/Folder/whatever... )? Delete that and type 

`cmd`

This will open a console where you type 

`python PixivUtil2.py`

which will launch the program. You will do this everytime to launch the program. I think theres a better way to compile it. If anyone know, please comment.

You need to run the program, get it to make you a config file (config.ini), then close it. Now you need your Cookie PHPSESSID thing, from BOTH pixiv and fanbox if you want to use both. Find those id numbers, and manually paste them into your config file, which you can open with notepad or notepad++. You have a personalized code. All this is listed on the FAQ on the main page. While you are in there, you can also redirect the files to be installed in the location of the original pixivutil folder. 

For example, this is what mine looks like

[Authentication]
username = this is blank for me actually. I just noticed this writing this terrible guide. So try leaving it blank.

password = So was this, I have nothing there, but it still works. Maybe it's readingmy original one in the first pixivutil install?

cookie = You need actual stuff here GET THIS FROM YOUR PIXIV ACCOUNT. Go to that site and open the web dev tools. Listed in the FAQ for this program. Read the readme on the main page!

cookieFanbox = DO THE SAME FOR YOUR FANBOX ACCOUNT. Go to the fanbox site, and do the same thing. Web dev tools.

refresh_token = this is also blank. Odd

cf_clearance = <cf_clearance COOKIE FROM BROWSER> (<---this is what it actual says for me, copy this if it aint there)

cf_bm = <__cf_bm COOKIE FROM BROWSER> (<---this is what it actual says for me, copy this if it aint there)

And, even after you do all that, some people have issues with Cloudflare. I hopped to a japanese VPN and it stops giving me issues. You might have to do the same. 

Hopefully this works good luck.

Again, not a developer, or someone who gets this shit, just a normie bumbling around in the dark looking at 15 tabs trying to figure this out.

**Comment by biggestsonicfan** [comment]
I'm going to do what I can to, after Christmas, look into fixing the login by adding [cloudscraper](https://github.com/VeNoMouS/cloudscraper) as a dependency to requirements to end the cookie nonsense for our less-than-technically-inclined users.

**Comment by GrasSchlammPferd** [comment]
> > **This Sucks**
> > As a **Normal User with no experience in this Shit**, I will probably be able to delete this nice tool at some point, because it has become unusable.
> > ☕
> 
> I am also a normal user, you need to do several things differently to get fanbox working properly. Find half an hour of time, sit down with some tea, and get ready to use a command window and type stuff. I AM NOT an expert or even familiar with this stuff. Please take this all with a grain of salt and correct me where I get things wrong.
> 
> First, you need to update the requirements as listed above. This means open cmd or powershell and running the commands listed above here on this page. if you get an error, just google the error. In my case, I had to run a command to uninstall the old ones I think, manually paste PixivUtil2's requirement.txt list the windows folder where your computer looks for it, then run the install command again until it cleared. I was just googling stuff and trying things listed here until it worked. Take a second and try doing that until you get the requirements updated.
> 
> https://github.com/Nandaka/PixivUtil2/blob/master/requirements.txt Download this file obviously.
> 
> You can't use the exe in the normal release, you have to use: https://github.com/user-attachments/files/16531982/PixivUtil.zip
> 
> Here's how: [#1361 (comment)](https://github.com/Nandaka/PixivUtil2/issues/1361#issuecomment-2275990761) Unzip the folder, open it. You should have a folder named PixivUtil2-20240703. This will be where you actually launch the program for now. Forget the old one until we get a proper update.
> 
> Make sure you have python installed. Google python and install it.
> 
> Replace the two files listed here in the folder named PixivUtil2-20240703 [#1361 (comment)](https://github.com/Nandaka/PixivUtil2/issues/1361#issuecomment-2273943969)
> 
> AND make these changes listed here to the same folder [#1352 (comment)](https://github.com/Nandaka/PixivUtil2/issues/1352#issuecomment-2227002669)
> 
> Then go to the address window of that folder you just downloaded, unzipped, and have been editing this whole time named PixivUtil2-20240703. You see the address window where normally it's just the drive location (E:/User/Folder/whatever... )? Delete that and type
> 
> `cmd`
> 
> This will open a console where you type
> 
> `python PixivUtil2.py`
> 
> which will launch the program. You will do this every time to launch the program. I think there's a better way to compile it. If anyone knows, please comment.
> 
> You need to run the program, get it to make you a config file (config.ini), then close it. Now you need your Cookie PHPSESSID thing, from BOTH pixiv and fanbox if you want to use both. Find those id numbers, and manually paste them into your config file, which you can open with notepad or notepad++. You have a personalized code. All this is listed on the FAQ on the main page. While you are in there, you can also redirect the files to be installed in the location of the original pixivutil folder.
> 
> For example, this is what mine looks like
> 
> [Authentication] username = this is blank for me actually. I just noticed this writing this terrible guide. So try leaving it blank.
> 
> password = So was this, I have nothing there, but it still works. Maybe it's readingmy original one in the first pixivutil install?
> 
> cookie = You need actual stuff here GET THIS FROM YOUR PIXIV ACCOUNT. Go to that site and open the web dev tools. Listed in the FAQ for this program. Read the readme on the main page!
> 
> cookieFanbox = DO THE SAME FOR YOUR FANBOX ACCOUNT. Go to the fanbox site, and do the same thing. Web dev tools.
> 
> refresh_token = this is also blank. Odd
> 
> cf_clearance = <cf_clearance COOKIE FROM BROWSER> (<---this is what it actual says for me, copy this if it aint there)
> 
> cf_bm = <__cf_bm COOKIE FROM BROWSER> (<---this is what it actual says for me, copy this if it aint there)
> 
> And, even after you do all that, some people have issues with Cloudflare. I hopped to a japanese VPN and it stops giving me issues. You might have to do the same.
> 
> Hopefully this works good luck.
> 
> Again, not a developer, or someone who gets this shit, just a normie bumbling around in the dark looking at 15 tabs trying to figure this out.
> 
> Edit: fixed some grammar and found this #1342
> 
> biggestsonicfan and nomakewan have generously explained how to update the program with the proper User Agent, and how to reset your cookies for both Pixiv and fanbox to circumvent the needing of a VPN. They also pointed out that for the laymen with no programming experience, this is definitely wading into relatively deep waters and relying on others to take the time to help you, so temper your impatience and frustration with the knowledge that getting the program to work is possible for now, as long as you actually READ both the open and closed issues related to your problem. I want to thank them both.
> 
> Second edit. I've having issues with the cookies. Someone suggested changing Pixiv language to japanese in the account settings. Tried that among other things. Got it working again.

@LordBarsdale I managed to get everything working but it's trying to download to the system32 folder, which it doesn't have permission.

I tried changing the config.ini's downloadListDirectory to a folder I set up but the error persists. Any clues?

**Comment by biggestsonicfan** [comment]
> I'm going to do what I can to, after Christmas, look into fixing the login by adding [cloudscraper](https://github.com/VeNoMouS/cloudscraper) as a dependency to requirements to end the cookie nonsense for our less-than-technically-inclined users.

So cloudscraper was already a dependency for another routine, and I got as far as the login screen to input credentials, but I couldn't pass the invisible cloudflare badge. I tried [stealthenium](https://github.com/markmelnic/stealthenium) with [SeleniumBase](https://github.com/seleniumbase/SeleniumBase) which has clicking captcha support, but I could get past the login screen to an infinite amount of click-captchas.

I could continue to research, but literally it will be harder for any of you to keep the chrome driver current with updates and obfuscated and get past the finite number of click captchas than it is to copy and paste a few cookie values.

**Comment by JeurseLightYokai** [comment]
followed everything you said @GrasSchlammPferd but now I'm stuck with "can't multiply sequence by non-int of type 'str'" error...

**Comment by JeurseLightYokai** [comment]
I've been stuck with hundreds of errors with this program for over a year now, reading issues and testing things for probably more than a hundred of hours in total and keeps hitting new errors after the previous one is solved. Being patient is a thing yeah but this is becoming really annoying.

**Comment by poyazi** [comment]
@krax11 
Hi, I've submitted #1404 , which removes unnecessary `.decode('utf-8')` calls when converting BeautifulSoup objects to strings. Since your issue seems to be caused by the same decode issue, this PR might resolve it.

Could you test with the latest master version and let me know if the issue persists?

**Comment by JeurseLightYokai** [comment]
@poyazi 
Yes, I confirm your fix does work on the latest version. thank you so much for your help !
@LordBarsdale 
Thank you for your help as well ! 
with all that the program finally works after a long period of unsuccessful fixing.

**Comment by BrazenwePARL** [comment]
I need help here btw
Traceback (most recent call last):
  File "D:\PixivUtil2-20240703\PixivUtil2.py", line 20, in <module>
    import PixivBatchHandler
  File "D:\PixivUtil2-20240703\PixivBatchHandler.py", line 14, in <module>
    import PixivUtil2
  File "D:\PixivUtil2-20240703\PixivUtil2.py", line 38, in <module>
    colorama.init()
    ^^^^^^^^
NameError: name 'colorama' is not defined

**Comment by JeurseLightYokai** [comment]
update: error shows up again after a day with no issue.

Traceback (most recent call last):
  File "C:\PixivUtil2-master\20240703\PixivUtil2.py", line 1745, in main
    np_is_valid, op_is_valid, selection = main_loop(ewd, op_is_valid, selection, np_is_valid, args, options)
  File "C:\PixivUtil2-master\20240703\PixivUtil2.py", line 1471, in main_loop
    menu_fanbox_download_by_post_id(op_is_valid, args, options)
  File "C:\PixivUtil2-master\20240703\PixivUtil2.py", line 971, in menu_fanbox_download_by_post_id
    post = __br__.fanboxGetPostById(post_id)
  File "C\PixivUtil2-master\20240703\PixivBrowserFactory.py", line 1085, in fanboxGetPostById
    js = self.fanboxGetPostJsonById(post_id)
  File "C:\PixivUtil2-master\20240703\PixivBrowserFactory.py", line 1094, in fanboxGetPostJsonById
    self.fanbox_is_logged_in()
  File "C:\PixivUtil2-master\20240703\PixivBrowserFactory.py", line 448, in fanbox_is_logged_in
    if not self.fanboxLoginUsingCookie(self._config.cookieFanbox):
  File "C:\PixivUtil2-master\20240703\PixivBrowserFactory.py", line 429, in fanboxLoginUsingCookie
    fanboxErrorPage = parsed.decode('utf-8')
  File "C:\Local\Programs\Python\Python310\lib\site-packages\bs4\__init__.py", line 1141, in decode
    return prefix + super(BeautifulSoup, self).decode(
  File "C:\Local\Programs\Python\Python310\lib\site-packages\bs4\element.py", line 2435, in decode
    piece = self._indent_string(
  File "C:\Local\Programs\Python\Python310\lib\site-packages\bs4\element.py", line 2520, in _indent_string
    space_before = formatter.indent * indent_level
TypeError: can't multiply sequence by non-int of type 'str'

**Comment by sk19vn** [comment]
I have tried every fix possible but nothing is helping. Even using the source code and all of the pip nonsense I can't even get the python version to run. Every time I attempt to open it by clicking on it it closes immediately. Trying to open it in cmd just leaves me with:
Traceback (most recent call last):
  File "C:\PixivUtil2-20240703\PixivUtil2.py", line 16, in <module>
	import colorama
ModuleNotFoundError: No module named 'colorama'

Every attempt to remove and reinstall just isn't working.

**Comment by sk19vn** [comment]
I have been following every part of this thread and am trying to use the source code variation because fanbox isn't working currently. I'm getting this crazy error currently:

Error at doLogin(): (<class 'TypeError'>, TypeError("can't multiply sequence by non-int of type 'str'"), <traceback object at 0x000002227B32EF80>)
Traceback (most recent call last):
  File "C:\PixivUtil2-20240703\PixivUtil2.py", line 1527, in doLogin
    result = __br__.loginUsingCookie()
  File "C:\PixivUtil2-20240703\PixivBrowserFactory.py", line 353, in loginUsingCookie
    parsed_str = str(parsed.decode('utf-8'))
                     ~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\User\AppData\Local\Programs\Python\Python313\Lib\site-packages\bs4\__init__.py", line 1141, in decode
    return prefix + super(BeautifulSoup, self).decode(
                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        indent_level, eventual_encoding, formatter, iterator
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "C:\Users\User\AppData\Local\Programs\Python\Python313\Lib\site-packages\bs4\element.py", line 2435, in decode
    piece = self._indent_string(
        piece, indent_level, formatter, indent_before, indent_after
    )
  File "C:\Users\User\AppData\Local\Programs\Python\Python313\Lib\site-packages\bs4\element.py", line 2520, in _indent_string
    space_before = formatter.indent * indent_level
                   ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~
TypeError: can't multiply sequence by non-int of type 'str'
